new/egeria-app.container.js

```diff
@@ -89,6 +68,37 @@ class EgeriaApp extends PolymerElement {
     ];
   }
 
+  /**
+   * performes requests needed by authenticated page
+   * @private
+   */
+  _initialRequiredAuthenticatedRequest() {
+    Promise.all([
+      egeriaFetch(`/api/public/app/info`),
+      egeriaFetch(`/api/users/components`),
+      egeriaFetch(`/api/users/current`),
+      egeriaFetch(`/api/users/roles`)
+    ]).then((responses) => {
+      let [ appInfo, components, currentUser, roles ] = responses;
+
+      this.components = components;
+      this.currentUser = currentUser;
+      this.roles = roles;
+      this.appInfo = appInfo;
+
+      this.isLoggedIn = currentUser !== null && currentUser.status === 200;
+
+      const hasComponents = components !== null && components.status === 200;
+      const hasCurrentUser = currentUser !== null && currentUser.status === 200;
+      const hasRoles = roles !== null &&  roles.status === 200;
+      const hasAppInfo = appInfo !== null && appInfo.status === 200;
+
+      if(![hasComponents, hasCurrentUser, hasRoles, hasAppInfo].includes(false)) {
+        this.isLoading = true;
+      }
```

Comment:
an else is not needed here as it was previously?

Reply:
not needed as it was initialised with false. 
But as long I extracted as a separate method, I should use it, or even better assign it directly. 
No need to have an if  statement only to assign true or false. :)